packages/compass-collection/src/modules/tabs.ts

```diff
@@ -182,56 +183,80 @@ const reducer: Reducer<CollectionTabsState> = (
       tabs: newTabs,
     };
   }
+  if (action.type === CollectionTabsActions.CollectionRenamed) {
+    const { tabs, activeTabIndex } = action;
+
+    const activeTabId = tabs[activeTabIndex]?.id;
```

Comment:
Reducer should be responsible for deriving new state from the old one and action, so in this case instead of passing active tab index from the action, you can do:

```suggestion
const activeTabIndex = getActiveTabIndex(state);
const activeTabId = tabs[activeTabIndex]?.id ?? null;
```